plugins/logging/pkg/gateway/drivers/management/cluster_driver.go

```diff
@@ -17,6 +17,11 @@ type ClusterDriver interface {
 	UpgradeAvailable(ctx context.Context, opniVersion string) (bool, error)
 	DoUpgrade(ctx context.Context, opniVersion string) error
 	GetStorageClasses(context.Context) ([]string, error)
+	CreateOrUpdateSnapshot(ctx context.Context, snapshot *loggingadmin.Snapshot) error
```

Comment:
I prefer to separate them out just in case there's some additional logic required

Reply:
If the methods are all the same, then it would be much easier to embed the server here like this:
```go
type ClusterDriver interface {
  loggingadmin.LoggingAdminV2Server

  // other methods here
}}
```
This makes the ClusterDriver interface easier to implement for multiple drivers. Otherwise, you need to duplicate the argument translation logic for each driver. If there is additional logic required from what's available in the grpc server interface, that would suggest the contents of the request/response types need to be adjusted.

Reply:
I don't think that is the correct abstraction.  The argument translation is specific to the ClusterDriver implementation.  I also think that the cluster driver should not care about Empty for protobuf and the error translation to protobuf should be done in the implementation of the LoggingAdminServer.  This separation also allows for where the logic is different at the LoggingAdminServer level, as in the CreateOrUpdateOpensearchCluster or the GetStatus.